compiler/rustc_session/src/options.rs

```diff
@@ -670,6 +671,18 @@ mod parse {
         }
     }
 
+    pub(crate) fn parse_frame_pointer(slot: &mut FramePointer, v: Option<&str>) -> bool {
+        let mut boolish = false;
+        let mut is_parsed = parse_bool(&mut boolish, v);
+        if boolish & is_parsed {
+            *slot = FramePointer::Always;
+        } else if v == Some("non-leaf") {
+            is_parsed = true;
+            *slot = FramePointer::NonLeaf;
+        };
```

Comment:
I think this would be a bit cleaner as a `match`.

Reply:
Sure!